src/Plivo/Resources/Message/MessageInterface.php

```diff
@@ -159,42 +160,91 @@ public function __call($name_of_method, $arguments)
             }
 
             $response = $this->client->update($this->uri, $arguments);
+
+            $responseContents = $response->getContent();
+
+                if (!array_key_exists("error", $responseContents))
+                {
+                    if (array_key_exists("invalid_number", $responseContents))
+                    {
+                        return new MessageCreateResponse($responseContents['message'], $responseContents['message_uuid'], $responseContents['api_id'], $response->getStatusCode() , $responseContents['invalid_number']);
+                    }
+                    else
+                    {
+                        return new MessageCreateResponse($responseContents['message'], $responseContents['message_uuid'], $responseContents['api_id'], $response->getStatusCode() , []);
+                    }
+                }
+                else
+                {
+                    throw new PlivoResponseException($responseContents['error'], 0, null, $response->getContent() , $response->getStatusCode());
+                }
         }
-        else
+
+        elseif($name_of_method == 'create' and count($arguments)>1)
         {
-            $src = $arguments[0];
+            if (empty($arguments[0]))
+            {   
+                $src = null;
+            }
+            else
+            {
+                $src = $arguments[0];
+            }
+            
             $dst = $arguments[1];
-            $text = $arguments[2];
-            $optionalArgs = $arguments[3];
-            $powerpackUUID = $arguments[4];
+            
+            if (empty($arguments[2]) || is_null($arguments[2]))
+            {   
+                $text = null;
```

Comment:
what case it should be null
?

Reply:
@narayana-plivo  If the type is mms, the text is treated as an optional parameter.